# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1219168023 14400
#      Tue Aug 19 13:47:03 2008 -0400
# Branch release-3-0-x
# Node ID 2994f58f204ef6ab2484ed7eb07b3d835a7316a2
# Parent  cf86d5c634dd8c656070a25c6e17eed31d827eff
parse.y (Feval): Return value produced by evaluating CATCH string

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,12 @@
+2008-08-19  John W. Eaton  <jwe@octave.org>
+
+	* parse.y (Feval): Return value produced by evaluating CATCH string.
+
 2008-08-19  Jaroslav Hajek <highegg@gmail.com>
 
 	Version 3.0.2 released.
 
 2008-08-19  Jaroslav Hajek  <highegg@m61s04>
 
 	* DLD-FUNCTIONS/rand.cc (Frandp): Fix typo in a test.
 
diff --git a/src/parse.y b/src/parse.y
--- a/src/parse.y
+++ b/src/parse.y
@@ -4012,32 +4012,32 @@ eval ('error (\"This is a bad example\")
 	  buffer_error_messages++;
 	}
 
       int parse_status = 0;
 
       octave_value_list tmp = eval_string (args(0), nargout > 0,
 					   parse_status, nargout);
 
-      if (nargout > 0)
-	retval = tmp;
-
       if (nargin > 1 && (parse_status != 0 || error_state))
 	{
 	  error_state = 0;
 
 	  // Set up for letting the user print any messages from
 	  // errors that occurred in the first part of this eval().
 
 	  buffer_error_messages--;
 
-	  eval_string (args(1), 0, parse_status, nargout);
-
-	  retval = octave_value_list ();
+	  tmp = eval_string (args(1), nargout > 0, parse_status, nargout);
+
+	  if (nargout > 0)
+	    retval = tmp;
 	}
+      else if (nargout > 0)
+	retval = tmp;
 
       unwind_protect::run_frame ("Feval");
     }
   else
     print_usage ();
 
   return retval;
 }
